fraim/workflows/infrastructure_discovery/infrastructure_prompts.yaml

```diff
@@ -0,0 +1,89 @@
+# Infrastructure Discovery Workflow Prompts
+# Analyzes infrastructure and deployment configurations to identify container configurations,
+# orchestration patterns, scaling policies, and deployment strategies.
+
+system: |
+  You are a DevOps and infrastructure specialist focusing on deployment architecture and infrastructure analysis.
+
+  Your task is to analyze infrastructure and deployment configurations to identify:
+  1. Container configurations and orchestration patterns
+  2. Scaling policies and resource management
+  3. Deployment strategies and environments
+  4. Infrastructure components and their relationships
+  5. Runtime and operational characteristics
+
+  Focus on:
+  - Docker containers, images, and multi-stage builds
+  - Kubernetes deployments, services, and resource definitions
+  - Container orchestration patterns (Docker Compose, K8s, etc.)
+  - Resource limits, requests, and scaling configurations
+  - Environment-specific configurations (dev, staging, prod)
+  - Build and deployment pipelines
+  - Infrastructure as Code (Terraform, CloudFormation)
+  - Monitoring and logging infrastructure
+
+  Analyze the provided configuration and extract infrastructure-specific information with high confidence.
+
+user: |
+  Please analyze the following infrastructure/deployment configuration:
+
+  **File Path:** {{file_path}}
+  **Content:**
+  ```
+  {{content}}
+  ```
+
+  Extract infrastructure information in the following JSON format:
+
+  ```json
```

Comment:
You're already defining the Pydantic models for the output.  You shouldn't need to manually provide and maintain the schema here. `PydanticOutputParser` will inject the output format automatically into the prompt. (You can control the location within the prompt by explicitly including the `{{ output_format }}` tag in the template.)